File: gherkin/javascript/src/ITokenMatcher.ts
import Token from './Token'

export default interface ITokenMatcher {
  reset(): void

  match_TagLine(token: Token): boolean

  match_FeatureLine(token: Token): boolean

  match_ScenarioLine(token: Token): boolean

  match_BackgroundLine(token: Token): boolean

  match_ExamplesLine(token: Token): boolean

  match_RuleLine(token: Token): boolean

  match_TableRow(token: Token): boolean

  match_Empty(token: Token): boolean

  match_Comment(token: Token): boolean

  match_Language(token: Token): boolean

  match_DocStringSeparator(token: Token): boolean

  match_EOF(token: Token): boolean

  match_StepLine(token: Token): boolean

  match_Other(token: Token): boolean
}

File: gherkin/dart/lib/src/parser/ITokenMatcher.dart
import 'package:gherkin/language.dart';

abstract class ITokenMatcher
{
  bool match_EOF(Token token);
  bool match_Empty(Token token);
  bool match_Comment(Token token);
  bool match_TagLine(Token token);
  bool match_FeatureLine(Token token);
  bool match_RuleLine(Token token);
  bool match_BackgroundLine(Token token);
  bool match_ScenarioLine(Token token);
  bool match_ExamplesLine(Token token);
  bool match_StepLine(Token token);
  bool match_DocStringSeparator(Token token);
  bool match_TableRow(Token token);
  bool match_Language(Token token);
  bool match_Other(Token token);
  void reset();
}
